File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    # brown=2*w+2(h-2)
    # yellow=(w-2)*(h-2)

    for w in range(3, 2500):
        for h in range(3, 2500):
            if yellow == (w - 2) * (h - 2) and brown == 2 * w + 2 * (h - 2):
                return [max(w,h),min(w,h)]

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    // brown = 2w + 2(h - 2) -> w = b / 2 - h + 2
    // yellow = (w - 2) * (h - 2) -> w = y / (h - 2) + 2

    let h = 0;
    while (brown / 2 - h !== yellow / (h - 2)) h++;
    return [brown / 2 - h + 2, h];
}